Lib/test/test_struct.py

```diff
@@ -723,23 +723,34 @@ def test_issue35714(self):
                 struct.calcsize(s)
 
     @support.cpython_only
-    def test_issue45034_unsigned(self):
-        _testcapi = import_helper.import_module('_testcapi')
-        error_msg = f'ushort format requires 0 <= number <= {_testcapi.USHRT_MAX}'
-        with self.assertRaisesRegex(struct.error, error_msg):
-            struct.pack('H', 70000)  # too large
-        with self.assertRaisesRegex(struct.error, error_msg):
-            struct.pack('H', -1)  # too small
-
-    @support.cpython_only
-    def test_issue45034_signed(self):
-        _testcapi = import_helper.import_module('_testcapi')
-        error_msg = f'short format requires {_testcapi.SHRT_MIN} <= number <= {_testcapi.SHRT_MAX}'
-        with self.assertRaisesRegex(struct.error, error_msg):
-            struct.pack('h', 70000)  # too large
-        with self.assertRaisesRegex(struct.error, error_msg):
-            struct.pack('h', -70000)  # too small
-
+    def test_issue92848(self):
+        def test_error_msg(prefix, int_type, is_unsigned):
+            fmt_str = prefix + int_type
+            size = struct.calcsize(fmt_str)
+            if is_unsigned:
+                max_ = 2 ** (size * 8) - 1
+                min_ = 0
+            else:
+                max_ = 2 ** (size * 8 - 1) - 1
+                min_ = -2 ** (size * 8 - 1)
+            error_msg = f"'{int_type}' format requires {min_} <= number <= {max_}"
+            for value in [int(-1e50), min_ - 1, max_ + 1, int(1e50)]:
+                with self.assertRaisesRegex(struct.error, error_msg):
+                    struct.pack(fmt_str, value)
+
+        for int_type in 'BHILQ':
```

Comment:
It would be better to use subtests here.

Reply:
Added. Thank you for your suggestion.